### PR TITLE
JAMES-3497 Various dependencies updates

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java [hunk 1]
```diff
@@ -141,16 +142,16 @@ public void configure(HierarchicalConfiguration<ImmutableNode> configuration) th
         List<Name> sPaths = new ArrayList<>();
         if (autodiscover) {
             LOGGER.info("Autodiscovery is enabled - trying to discover your system's DNS Servers");
-            String[] serversArray = ResolverConfig.getCurrentConfig().servers();
+            List<InetSocketAddress> serversArray = ResolverConfig.getCurrentConfig().servers();
             if (serversArray != null) {
-                for (String aServersArray : serversArray) {
-                    dnsServers.add(aServersArray);
+                for (InetSocketAddress aServersArray : serversArray) {
+                    dnsServers.add(aServersArray.getHostName());
```
Comment (rouazana) [inline server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java:148]: are you sure you want the hostname here?

Comment (rouazana) [reply server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java:148]: in doubt I would have used `getHostString`

Comment (chibenwa) [reply server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java:148]: `                    dnsServers.add(aServersArray.getAddress().toString());`

Gets the tests failing:

```
[ERROR] o.a.j.d.d.DNSJavaService - DNS service could not be initialized.  The DNS servers specified are not recognized hosts.
java.net.UnknownHostException: /127.0.0.53: Name or service not known
	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
```

I imagine my above snipset attemps a resolution.

`dnsServers.add(aServersArray.getHostString());` also works.

What do you prefer?

Comment (chibenwa) [reply server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java:148]: (please note that nothing forbids to pass an IP address as an hostname)

Comment (rouazana) [reply server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java:148]: if you pass an IP address getHostName will attempt a resolution. Do we want that?

Comment (chibenwa) [reply server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java:148]:
```

String | getHostString() Returns the hostname, or the String form of the address if it  doesn't have a hostname (it was created using a literal).
```

Let's go for that option.

**Comment by chibenwa** [comment]
```
[aefcc82f3cf71375598b2e66695906cc8b675c44] [ERROR] Failures: 
[aefcc82f3cf71375598b2e66695906cc8b675c44] [ERROR]   MemoryBlobStoreDAOTest.saveShouldThrowWhenNullInputStream 
[aefcc82f3cf71375598b2e66695906cc8b675c44] Expecting code to raise a throwable.
[aefcc82f3cf71375598b2e66695906cc8b675c44] [INFO] 
[aefcc82f3cf71375598b2e66695906cc8b675c44] [ERROR] Tests run: 199, Failures: 1, Errors: 0, Skipped: 0
```

**Comment by chibenwa** [comment]
`[54/247]`

**Review by rouazana — APPROVED** [review]

**Comment by chibenwa** [comment]
```
[INFO] Building Apache James :: Server :: Guice :: Configuration 3.6.0-SNAPSHOT [118/247]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project james-server-guice-configuration: There are test failures.
```

**Comment by chibenwa** [comment]
`[INFO] Building Apache James :: Server :: Guice Common 3.6.0-SNAPSHOT [144/247]`

**Comment by chibenwa** [comment]
`[ERROR]   GetMessagesMethodTest.processShouldReturnPropertyFilterWhenFilteringHeadersRequested:458 » InvalidDefinition`

File: pom.xml [hunk 2]
```diff
@@ -2527,7 +2527,7 @@
             </dependency>
             <dependency>
                 <groupId>org.awaitility</groupId>
-                <artifactId></artifactId>
+                <artifactId>awaitility</artifactId>
                 <version>3.1.6</version>
```
Comment (Arsnael) [inline pom.xml]: still 3.1.6 here, not 4.0.3

**Comment by chibenwa** [comment]
```
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] SessionRoutesTest:
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] get
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] - should return OK status
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] options
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] - should return OK status
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] get .well-known/jmap
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] WARNING: An illegal reflective access operation has occurred
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v9.Java9 (file:/root/.m2/repository/org/codehaus/groovy/groovy/3.0.7/groovy-3.0.7.jar) to constructor java.lang.AssertionError(java.lang.String)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v9.Java9
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] WARNING: All illegal access operations will be denied in a future release
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] - should redirect *** FAILED ***
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   java.lang.AssertionError: 1 expectation failed.
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8] Expected status code <308> but was <200>.
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:72)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:59)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:84)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
[9c8455cb5ffe1c99844f3218658deb80b4c3afc8]   ...
```

**Comment by chibenwa** [comment]
```
[INFO] Building Apache James :: Server :: Web Admin :: Core 3.6.0-SNAPSHOT [167/247]

[48ed709ba152d4a07238b39b2ff31855023b552b] [ERROR] parseShouldThrowOnNullInput  Time elapsed: 0.012 s  <<< FAILURE!
[48ed709ba152d4a07238b39b2ff31855023b552b] java.lang.AssertionError: 
[48ed709ba152d4a07238b39b2ff31855023b552b] 
[48ed709ba152d4a07238b39b2ff31855023b552b] Expecting actual throwable to be an instance of:
[48ed709ba152d4a07238b39b2ff31855023b552b]   java.lang.NullPointerException
[48ed709ba152d4a07238b39b2ff31855023b552b] but was:
[48ed709ba152d4a07238b39b2ff31855023b552b]   org.apache.james.webadmin.utils.JsonExtractException: argument "content" is null
```

**Comment by chibenwa** [comment]
```
[INFO] Building Apache James :: Server :: App 3.6.0-SNAPSHOT          [214/247]

[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR] springShouldLoadAndAddOnlyOneQuotaUpdaterListener  Time elapsed: 0.618 s  <<< FAILURE!
[979f542ff861c3e40bd35b4c15e4161543c14e03] org.opentest4j.MultipleFailuresError: 
[979f542ff861c3e40bd35b4c15e4161543c14e03] Multiple Failures (2 failures)
[979f542ff861c3e40bd35b4c15e4161543c14e03] 	org.springframework.beans.factory.xml.XmlBeanDefinitionStoreException: Line 86 in XML document from class path resource [META-INF/org/apache/james/spring-server.xml] is invalid; nested exception is org.xml.sax.SAXParseException; lineNumber: 86; columnNumber: 114; cvc-complex-type.3.2.2: Attribute 'createConnector' is not allowed to appear in element 'camel:jmxAgent'.
[979f542ff861c3e40bd35b4c15e4161543c14e03] 	java.lang.NullPointerException: <no message>
[979f542ff861c3e40bd35b4c15e4161543c14e03] 
[979f542ff861c3e40bd35b4c15e4161543c14e03] [INFO] Running org.apache.james.app.spring.JamesAppSpringMainTest
[979f542ff861c3e40bd35b4c15e4161543c14e03] SLF4J: Class path contains multiple SLF4J bindings.
[979f542ff861c3e40bd35b4c15e4161543c14e03] SLF4J: Found binding in [jar:file:/root/.m2/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[979f542ff861c3e40bd35b4c15e4161543c14e03] SLF4J: Found binding in [jar:file:/root/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.14.0/log4j-slf4j-impl-2.14.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[979f542ff861c3e40bd35b4c15e4161543c14e03] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
[979f542ff861c3e40bd35b4c15e4161543c14e03] SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.661 s <<< FAILURE! - in org.apache.james.app.spring.JamesAppSpringMainTest
[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR] testServer  Time elapsed: 0.652 s  <<< ERROR!
[979f542ff861c3e40bd35b4c15e4161543c14e03] org.springframework.beans.factory.xml.XmlBeanDefinitionStoreException: Line 86 in XML document from class path resource [META-INF/org/apache/james/spring-server.xml] is invalid; nested exception is org.xml.sax.SAXParseException; lineNumber: 86; columnNumber: 114; cvc-complex-type.3.2.2: Attribute 'createConnector' is not allowed to appear in element 'camel:jmxAgent'.
[979f542ff861c3e40bd35b4c15e4161543c14e03] 	at org.apache.james.app.spring.JamesAppSpringMainTest.testServer(JamesAppSpringMainTest.java:27)
[979f542ff861c3e40bd35b4c15e4161543c14e03] Caused by: org.xml.sax.SAXParseException: cvc-complex-type.3.2.2: Attribute 'createConnector' is not allowed to appear in element 'camel:jmxAgent'.
[979f542ff861c3e40bd35b4c15e4161543c14e03] 	at org.apache.james.app.spring.JamesAppSpringMainTest.testServer(JamesAppSpringMainTest.java:27)
[979f542ff861c3e40bd35b4c15e4161543c14e03] 
[979f542ff861c3e40bd35b4c15e4161543c14e03] [INFO] 
[979f542ff861c3e40bd35b4c15e4161543c14e03] [INFO] Results:
[979f542ff861c3e40bd35b4c15e4161543c14e03] [INFO] 
[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR] Failures: 
[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR]   JamesSpringContextTest.springShouldLoadAndAddOnlyOneQuotaUpdaterListener Multiple Failures (2 failures)
[979f542ff861c3e40bd35b4c15e4161543c14e03] 	org.springframework.beans.factory.xml.XmlBeanDefinitionStoreException: Line 86 in XML document from class path resource [META-INF/org/apache/james/spring-server.xml] is invalid; nested exception is org.xml.sax.SAXParseException; lineNumber: 86; columnNumber: 114; cvc-complex-type.3.2.2: Attribute 'createConnector' is not allowed to appear in element 'camel:jmxAgent'.
[979f542ff861c3e40bd35b4c15e4161543c14e03] 	java.lang.NullPointerException: <no message>
[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR] Errors: 
[979f542ff861c3e40bd35b4c15e4161543c14e03] [ERROR]   JamesAppSpringMainTest.testServer:27 » XmlBeanDefinitionStore Line 86 in XML d...
```

**Comment by chibenwa** [comment]
```
[INFO] Building Apache James :: Server :: Mailet :: Mock SMTP Server 3.6.0-SNAPSHOT [233/247]

[INFO] Apache James :: Server :: Mailet :: Mock SMTP Server FAILURE [  9.474 s]

[7cde2a12c2e7015976bba54df0d336d09f0849d0] [ERROR] Failures: 
[7cde2a12c2e7015976bba54df0d336d09f0849d0] [ERROR]   ConditionTest$JSONTest.jacksonShouldThrowWhenDeserializeContainsConditionWithoutMatchingValue:155 
[7cde2a12c2e7015976bba54df0d336d09f0849d0] Expecting actual throwable to be an instance of:
[7cde2a12c2e7015976bba54df0d336d09f0849d0]   com.fasterxml.jackson.databind.exc.InvalidDefinitionException
[7cde2a12c2e7015976bba54df0d336d09f0849d0] but was:
[7cde2a12c2e7015976bba54df0d336d09f0849d0]   com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `org.apache.james.mock.smtp.server.model.Condition$Builder`, problem: You need to specify a matchingValue with the contains operator
[7cde2a12c2e7015976bba54df0d336d09f0849d0]  at [Source: (String)"{"operator":"contains"}"; line: 1, column: 23]
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	at com.fasterxml.jackson.databind.exc.ValueInstantiationException.from(ValueInstantiationException.java:47)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	at com.fasterxml.jackson.databind.DeserializationContext.instantiationException(DeserializationContext.java:1907)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	at com.fasterxml.jackson.databind.DeserializationContext.handleInstantiationProblem(DeserializationContext.java:1260)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	...(90 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] [ERROR]   ConditionTest$JSONTest.jacksonShouldThrowWhenDeserializeMatchAllConditionWithMatchingValue:146 
[7cde2a12c2e7015976bba54df0d336d09f0849d0] Expecting actual throwable to be an instance of:
[7cde2a12c2e7015976bba54df0d336d09f0849d0]   com.fasterxml.jackson.databind.exc.InvalidDefinitionException
[7cde2a12c2e7015976bba54df0d336d09f0849d0] but was:
[7cde2a12c2e7015976bba54df0d336d09f0849d0]   com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `org.apache.james.mock.smtp.server.model.Condition$Builder`, problem: You should not specify a matchingValue with the matchAll operator
[7cde2a12c2e7015976bba54df0d336d09f0849d0]  at [Source: (String)"{"operator":"matchAll", "matchingValue":"matchme"}"; line: 1, column: 50]
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	at com.fasterxml.jackson.databind.exc.ValueInstantiationException.from(ValueInstantiationException.java:47)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	at com.fasterxml.jackson.databind.DeserializationContext.instantiationException(DeserializationContext.java:1907)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	at com.fasterxml.jackson.databind.DeserializationContext.handleInstantiationProblem(DeserializationContext.java:1260)
[7cde2a12c2e7015976bba54df0d336d09f0849d0] 	...(90 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
```

Very slowly moving forward!

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`Caused by: java.io.IOException: Failed to bind to /0.0.0.0:35315`

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[eb0c03f6d61e835039d6357be9786c8cc86bcccd] [INFO] Apache James :: Server :: JMAP RFC-8621 :: Memory Integration Testing FAILURE [03:51 min]
```

Rebase and fix the 35 Jackson error messages issues

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
(Just rebased)

**Pull request merged**